chore: steps to add API auth modes using CLI (#3271)
aws-amplify/docs: docs/lib/graphqlapi/fragments/native_common/authz/common.md

> For client authorization AppSync supports API Keys, Amazon IAM credentials, Amazon Cognito User Pools, and 3rd party OIDC providers. This is inferred from the `amplifyconfiguration.json` file when you call `Amplify.configure()`. You can configure auth modes for API using Amplify CLI or manual configuration.

Reviewer: ```suggestion
For client authorization AppSync supports API Keys, Amazon IAM credentials, Amazon Cognito User Pools, and 3rd party OIDC providers. This is inferred from the `amplifyconfiguration.json` file when you call `Amplify.configure()`. You can configure auth modes for an API using the Amplify CLI or manual configuration.
```

## Auth Modes

> #### API key

Reviewer 2: Please convert these to H3, so they show up on the right-hand nav

> API Key is the easiest way to setup and prototype your application with AWS AppSync. This means it is also prone to abuse since anyone can easily discover the API Key and make requests to your public service. To have authorization checks, use the other auth modes such as Cognito user pool or AWS IAM. API Key will expiry according to the expiry time set when provisioning AWS AppSync and will require extending it or creating a new one if needed. 

Reviewer 3: Should we mention here that the default expiry for APIKey is 7 days.

Author: yes, that's a good idea.

#### Amazon Cognito User Pools

> Amazon Cognito's user pool is most commonly used with AWS AppSync when adding authorization check on your API calls. If your application needs to interact with other AWS services besides AWS AppSync, such as Amazon S3, you will need to use AWS IAM credentials with Amazon Cognito's identity pools. Amplify CLI can automatically configure this for you and will also automatically use the authenticated user from user pools to federate with the identity pools to provide the AWS IAM credentials in the application. [See this for more information about the differences](https://aws.amazon.com/premiumsupport/knowledge-center/cognito-user-pools-identity-pools/). This allows you to have both user pool credentials for AWS AppSync and AWS IAM credentials for other AWS resources. You can learn more about Amplify Auth outlined in the [Accessing credentials section](~/lib/auth/access_credentials.md). 

Reviewer 2: ```suggestion
Amazon Cognito User Pools is most commonly used with AWS AppSync when adding authorization check on your API calls. If your application needs to interact with other AWS services besides AWS AppSync, such as Amazon S3, you will need to use AWS IAM credentials with Amazon Cognito Identity Pools. Amplify CLI can automatically configure this for you and will also automatically use the authenticated user from User Pools to federate with the Identity Pools to provide the AWS IAM credentials in the application. [See this for more information about the differences](https://aws.amazon.com/premiumsupport/knowledge-center/cognito-user-pools-identity-pools/). This allows you to have both User Pools' credentials for AWS AppSync and AWS IAM credentials for other AWS resources. You can learn more about Amplify Auth outlined in the [Accessing credentials section](~/lib/auth/access_credentials.md). 
```

#### IAM

> Amazon Cognito identity pools allows you to use credentials from AWS IAM in a mobile application. The Amplify CLI can automatically configure this for you.

Reviewer 2: ```suggestion
Amazon Cognito Identity Pools allows you to use credentials from AWS IAM in your app. The Amplify CLI can automatically configure this for you.
```

> Amazon Cognito identity pools allows you to use credentials from AWS IAM in a mobile application. The Amplify CLI can automatically configure this for you.

Reviewer 2: I'd also add the reasoning behind using IAM. Customers new to AWS would be overwhelmed by the technology terms but don't understand the use case necessarily.

> #### OIDC

Reviewer 2: For abbreviations use the full term first with the abbreviations in parentheses.

If you are using a 3rd party OIDC provider you will need to configure it and manage the details of token refreshes yourself. 

> ## Using Amplify CLI

Reviewer 2: Use Amplify CLI to configure authorization modes

> Amplify CLI can automatically configure the auth modes for you when running `amplify add auth` or `amplify update auth` if you want to change the auth mode.

Reviewer 2: this is incorrect. it sohuld be `amplify add api` or `amplify update api`

> After adding auth mode, you have to update API category to use the auth mode you just added by running:

Reviewer 2: please verify but I believe you can add the auth modes right from `amplify add/update api` because it'll just "redirect" into the auth flow.

<inline-fragment platform="ios" src="~/lib/graphqlapi/fragments/ios/authz/21_oidc.md"></inline-fragment>
<inline-fragment platform="android" src="~/lib/graphqlapi/fragments/android/authz/21_oidc.md"></inline-fragment>

> ### Multi-Auth

Reviewer 2: Configure multiple authorization modes
